Core/include/Acts/Utilities/BinAdjustment.hpp

```diff
@@ -141,12 +192,17 @@ BinUtility adjustBinUtility(const BinUtility& bu, const Surface& surface) {
     auto rBounds = dynamic_cast<const RadialBounds*>(&(surface.bounds()));
     // Return specific adjustment
     return adjustBinUtility(bu, *rBounds, surface.transform(GeometryContext()));
-  }
-
-  throw std::invalid_argument(
-      "Bin adjustment not implemented for this surface yet!");
+  } else if (surface.type() == Surface::Plane) {
+    // Cast to Rectangle bounds and return
+    auto rBounds = dynamic_cast<const RectangleBounds*>(&(surface.bounds()));
+    // Return specific adjustment
+    return adjustBinUtility(bu, *rBounds, surface.transform(GeometryContext()));
+  } else {
+    std::cout << " Surface type = " << surface.type() << std::endl;
 
```

Comment:
Please do not use `std::cout` in the code, if output it necessary, use a `ACTS_LOCAL_LOGGER(...)` macro, however, in this case I would suggest to remove the screen output entirely.